src/applications/disability-benefits/view-payments/tests/components/view-payments-lists/view-payments-lists.unit.spec.jsx

```diff
@@ -1,84 +1,117 @@
 import React from 'react';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+import { renderInReduxProvider } from 'platform/testing/unit/react-testing-library-helpers';
 import { expect } from 'chai';
-import { render } from '@testing-library/react';
-import sinon from 'sinon';
 import allPayments from '../../../reducers/index';
 import createCommonStore from 'platform/startup/store';
+import environment from 'platform/utilities/environment';
+import ViewPaymentsLists from '../../../components/view-payments-lists/ViewPaymentsLists.jsx';
+import {
+  payments,
+  emptyPaymentsReturned,
+  emptyPaymentsReceived,
+  emptyPaymentsResponse,
+} from '../../helpers';
 
-const store = createCommonStore(allPayments);
+const initialState = {
+  isLoading: false,
+  payments: null,
+  error: false,
+};
 
-import ViewPaymentsLists from '../../../components/view-payments-lists/ViewPaymentsLists.jsx';
+const store = createCommonStore(allPayments);
 
-import { payments } from '../../helpers';
+describe('View Payments Lists', () => {
+  const server = setupServer(
+    rest.get(
+      `${environment.API_URL}/v0/profile/payment_history`,
+      (req, res, ctx) => {
+        return res(
+          ctx.json({
+            data: {
+              attributes: payments,
+              metadata: [],
+            },
+          }),
+        );
+      },
+    ),
+  );
 
-describe.skip('View Payments Lists', () => {
-  it('renders View Payments Lists component', async () => {
-    const getAllPayments = sinon.spy();
-    const wrapper = render(
-      <ViewPaymentsLists
-        store={store}
-        isLoading={false}
-        error={false}
-        payments={payments}
-        getAllPayments={getAllPayments}
-      />,
+  const overrideServerWithOptions = payload => {
+    server.use(
+      rest.get(
+        `${environment.API_URL}/v0/profile/payment_history`,
+        (req, res, ctx) => {
+          return res.once(
+            ctx.json({
+              data: {
+                attributes: payload,
+                metadata: [],
+              },
+            }),
+          );
+        },
+      ),
     );
-    await expect(getAllPayments.toBeCalled);
-    await expect(wrapper.findByText(/Payments you received/)).to.exist;
-  });
+  };
 
-  it('shows a loading indicator when loading', () => {
-    const wrapper = render(<ViewPaymentsLists store={store} isLoading />);
+  before(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  after(() => server.close());
 
-    expect(wrapper.getByText(/Loading payment information.../)).to.exist;
+  it('renders View Payments Lists component with both tables', async () => {
+    const screen = renderInReduxProvider(<ViewPaymentsLists />, {
+      initialState,
+      allPayments,
+      store,
+    });
+    expect(await screen.findByText(/Loading payment information.../)).to.exist;
```

Comment:
In some of the VAOS tests we've stopped testing for loading indicators this way because it doesn't seem guaranteed that you'll get a render with the loading text before the mocked request is resolved. It's pretty rare that it causes problems, but thought I'd mention it in case you see it on Jenkins later.

Reply:
That's a good thought, since loading indicators only represent an intermediary state. I'll update.